vivisect/impemu/platarch/arm.py

```diff
@@ -212,7 +214,7 @@ def runFunction(self, funcva, stopva=None, maxhit=None, maxloop=None, tmode=None
                             logger.debug(repr(e))
                             break
                         except Exception as e:
-                            logger.warning("funcva: 0x%x opva: 0x%x:  %r   (%r) (in emumon prehook: %r)", funcva, starteip, op, e, self.emumon)
+                            logger.log(e_common.EMULOG, "funcva: 0x%x opva: 0x%x:  %r   (%r) (in emumon prehook: %r)", funcva, starteip, op, e, self.emumon)
```

Comment:
how about setting an emulator-specific log-level for this??
so here it says "logger.log(self.exception_level, "funcva.....")" but we set exception_level per-emulator instance.

Reply:
So I would be fine with exception levels being per emulator instances. That does work better.

But just for going forward, I just generally dislike treating the meta dictionary as a storage space for runtime behavior. That's not what it's there for and having this undocumented parameter float along through there is imo kinda terrible. It looks and acts about as janky as the old "create a class and then reach in and monkey with the internal class member variables" that just to be scattered around.

Reply:
glad we're in agreement on the logging.  

as for the other, i understand in concept... but are you suggesting that we not use emulation metadata for context during emulation?  that's the reason i plummed emulation.set/getMeta() through :)  the emulator never had that until this exact use-case. 

to be honest, i'm not sold on metadata being the location for that information either.  perhaps storing that info in a object-variable is better... but then you're either adding another field to an existing object (which is equally poor-form), or you're subclassing Emulator for each thing, which is suboptimal, since getEmulator() doesn't take arguments for what kind of emulator, so then you're either manually digesting an existing emulator into your new one, or you're doing `emu2.setEmuSnap(emu1.getEmuSnap())`

it's an ugly problem, and often this type of thing is for data necessary within an emulation monitor, which you want to be able to plug into any emulator.

Reply:
per offline conversation, i'll approve as is.  this PR has become far more than it was intended ;)
please fix this soon.  i neither want to miss legitimate exceptions nor see the results of emulating through garbage.